engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -227,15 +228,15 @@ - (instancetype)initWithName:(NSString*)labelPrefix
                  name:UIApplicationDidReceiveMemoryWarningNotification
                object:nil];
 
-#if APPLICATION_EXTENSION_API_ONLY
-  if (@available(iOS 13.0, *)) {
-    [self setUpSceneLifecycleNotifications:center];
-  } else {
+  if ([FlutterSharedApplication isAvailable]) {
     [self setUpApplicationLifecycleNotifications:center];
```

Comment:
Instead of the caller (in two places) needing to understand the extension and API limitations of this call, can the logic be pushed down into one `setUpLifecycleNotifications:` method?